v2/generator/generator.go

```diff
@@ -106,28 +107,50 @@ func (g *Generator) Generate(schema *models.Schema) error {
 	defer os.RemoveAll(g.tempDir)
 
 	// execute type modules
-	for _, mod := range g.typeModules {
-		for _, tbl := range schema.Types {
-			if err := g.ExecuteTemplate(mod, tbl.Name, tbl); err != nil {
+	for _, tbl := range schema.Types {
+		registry := NewPackageRegistry(g.pkg)
+		tds := &typeDataSet{
```

Comment:
I need to pass the package registry to the type template, so I defined `typeDataSet`.